internal/tree/treeconfig_test.go

```diff
@@ -0,0 +1,150 @@
+package tree
+
+import (
+	"fmt"
+	"slices"
+	"testing"
+
+	"github.com/relab/hotstuff"
+)
+
+// replicaIDs returns a slice of hotstuff.IDs from 1 to size.
+func replicaIDs(size int) []hotstuff.ID {
+	ids := make([]hotstuff.ID, size)
+	for i := range size {
+		ids[i] = hotstuff.ID(i + 1)
+	}
+	return ids
+}
+
+func TestCreateTree(t *testing.T) {
+	tests := []struct {
+		configurationSize int
+		id                hotstuff.ID
+		branchFactor      int
+		wantHeight        int
+	}{
+		{configurationSize: 10, id: 1, branchFactor: 2, wantHeight: 4},
+		{configurationSize: 21, id: 1, branchFactor: 4, wantHeight: 3},
+		{configurationSize: 21, id: 1, branchFactor: 3, wantHeight: 4},
+		{configurationSize: 43, id: 1, branchFactor: 5, wantHeight: 4},
+		{configurationSize: 43, id: 1, branchFactor: 4, wantHeight: 4},
+		{configurationSize: 111, id: 1, branchFactor: 10, wantHeight: 3},
+		{configurationSize: 111, id: 1, branchFactor: 3, wantHeight: 5},
+	}
+	for _, test := range tests {
+		tree := CreateTree(test.id, test.branchFactor, replicaIDs(test.configurationSize))
+		if tree.TreeHeight() != test.wantHeight {
+			t.Errorf("CreateTree(%d, %d, %d).GetTreeHeight() = %d, want %d",
+				test.configurationSize, test.id, test.branchFactor, tree.TreeHeight(), test.wantHeight)
+		}
+	}
+}
+
+func TestCreateTreeNegativeBF(t *testing.T) {
+	defer func() { _ = recover() }()
+	ids := []hotstuff.ID{1, 2, 3, 4, 5}
+	tree := CreateTree(1, -1, ids)
+	t.Errorf("CreateTree should panic, got %v", tree)
+}
+
+func TestCreateTreeInvalidID(t *testing.T) {
+	defer func() { _ = recover() }()
+	ids := []hotstuff.ID{1, 2, 3, 4, 5}
+	tree := CreateTree(10, 2, ids)
+	t.Errorf("CreateTree should panic, got %v", tree)
+}
+
+func TreeTreeMethods(t *testing.T) {
+	treeConfigTestData := []struct {
+		configurationSize int
+		id                hotstuff.ID
+		branchFactor      int
+		wantTreeHeight    int
+		wantChildren      []hotstuff.ID
+		wantSubTree       []hotstuff.ID
+		wantParent        hotstuff.ID
+		wantIsRoot        bool
+		wantReplicaHeight int
+		wantPeers         []hotstuff.ID
+	}{
+		{10, 1, 2, 4, []hotstuff.ID{2, 3}, []hotstuff.ID{2, 3, 4, 5, 6, 7, 8, 9, 10}, 1, true, 4, []hotstuff.ID{}},
```

Comment:
This table is hard to use/edit because the struct contains too many fields. Moreover, the fields should also be named on each line in the table (see examples on Line 27-33).

This test should be split up into separate tests for each function with its own anonymous `tests := struct[]{`:

```go
func TestTreeHeight  
func TestReplicaChildren  
func TestChildrenOf  
func TestParent  
func TestSubTree  
```
etc. 

By splitting up the tests, you also avoid long unlabeled table entries which are very hard to decode visually.